### DISCUSSION TITLE
Manually including content in zip from `dotnet lambda package`

### DISCUSSION THREAD

**Discussion by JohnYoungers** [aws/aws-lambda-dotnet#1776] [Q&A] [answered]
Our project references a nuget library that contains its own `.targets` file, which after build, adds files to the output directory.

Given these aren't technically part of the project, I think `dotnet lambda package` ignores them when it creates the zip file.  Is there an MSBuild hook, or a CLI option to specify adding additional content files?

**Comment by normj** [comment]
Not sure why they targets wouldn't be honored. The `dotnet lambda package` command ultimately does a `dotnet publish` command on the lambda project and then zips up whatever is in the output folder. Any way you can create a repo case?

**Reply by JohnYoungers** [reply]
My guess is it either doesn't really copy what's in the output folder, or it does it earlier than this target would run:

```
<Target Name="CopyContent" AfterTargets="Build">
	<Copy SourceFiles="$(MSBuildThisFileDirectory)..\content\myfile.yaml" DestinationFolder="$(OutDir)"  />
</Target>
```

For now I'm just adding the file as content instead:

```
<ItemGroup>
	<Content Include="$(MSBuildThisFileDirectory)..\content\myfile.yaml" CopyToOutputDirectory="PreserveNewest" />
</ItemGroup>
```

**Reply by normj** [reply, marked as answer]
That was an interesting issue. Like I said the tooling eventually does `dotnet publish --output <some-folder>`. That folder set by the `--output` is not the same as `$(OutDir)`. So in your example the `myfile.yaml` is copied to the build directory but not the publish folder. To get the file in the publish folder you would use `$(PublishDir)`.

**Reply by JohnYoungers** [reply]
I see; that makes sense: thank you for looking into this!
